js/currency-converter.js

```diff
@@ -0,0 +1,1071 @@
+
+function hide_inputs_by_year() {
+    if (get_currency_year() > 1970) {
+        /* Hide old UK currency inputs & show modern */
+        $("#currency-shillings-row").hide();
+        $("#currency-old-pence-row").hide();
+        $("#currency-new-pence-row").show();
+    }
+    else if (get_currency_year() <= 1970) {
+        /* Hide modern UK currency inputs & show old */
+        $("#currency-shillings-row").show();
+        $("#currency-old-pence-row").show();
+        $("#currency-new-pence-row").hide();
+    }
+}
+
+$("#currency-year").change(function () {
+    change_fieldset_text();
+    hide_inputs_by_year();
+
+});
+
+function change_fieldset_text() {
+    $('legend').text("Enter currency to show its purchasing power in " + get_currency_year());
+    $('#currency-submit').val("Show purchasing power in " + get_currency_year());
+}
+
+function build_blockquote(intro, century) {
+    return "<blockquote><p>" + intro + "</p>" +
+        "<p><cite><a href='./" + century + "-century.php'>Read more about the " + century + " century. </a></p></cite></blockquote>";
+}
+
+function currency_output() {
+    var user_inputs = get_user_inputs();
+    var currency_formula_return_values = currency_formula();
+
+    var century_preview = "";
+    var converted_money_string = currency_formula_return_values.money.toLocaleString('en-GB', {style: 'currency', currency: 'GBP'});
+
+    if (currency_formula_return_values.century != "21st") {
+            century_preview = build_blockquote(conversion_data.century_intros[user_inputs.century], user_inputs.century);
+    }
+
+    var HTML_output =
+
+        "<div class='currency' id='currency-result'><h1>In 2017, this is worth approximately: </h1>" + "<span id='currency-large-text'>" + converted_money_string + "</span>" +
+
+        "<h3>In " + user_inputs.year +", you could buy one of the following with " + currency_formula_return_values.bp_string + ": </h3>" +
+
+        build_currency_output_html("Horses", currency_formula_return_values.horses, "", "./img/horse.png") +
+
+        build_currency_output_html("Cows", currency_formula_return_values.cows, "", "./img/cow.png") +
+
+        build_currency_output_html("Wool", currency_formula_return_values.wool, "stones", "./img/ewe.png") +
+
+        build_currency_output_html("Wheat", currency_formula_return_values.wheat, "quarters", "./img/wheat.png") +
+
+        build_currency_output_html("Wages", currency_formula_return_values.wage, "days (skilled tradesman)", "./img/coinage.png") +
+
+        century_preview +
+        "</div>";
+
+    $("#currency-result").html(HTML_output);
+    $("#currency-result").fadeIn();
+}
+
+function build_currency_output_html(string, value, unit, img) {
+    return "<h4>" + " <img src='" + img + "'/>" + string + ": " + value + " " + unit + "</h4>";
+}
+
+$("#currency-form").submit(function (event) {
+    event.preventDefault(); // Prevent PHP fallback
+    window.location.hash = '';
+    window.location.hash = '#currency-result';
```

Comment:
There are two assignments to `window.location.hash` immediately following each other here.